File: src/problems/rts_buildorders/BAR_game/BAR_extractor_scripts/create_unit_enum.lua
local json = require("json")
local file = io.open("../Beyond-All-Reason/language/en/units.json", "r")
local content = file:read("*all")
file:close()

local desiredUnits = dofile("unitPaths.lua")

local language = json.decode(content)

print("enum BAR_UnitType {")

for unit, _ in pairs(language["units"]["names"]) do
    if desiredUnits[unit] then
        print("BAR_UnitType_" unit .. ",")
    end
end

print("BAR_UnitType_MAX\n};")

File: src/problems/rts_buildorders/BAR_game/BAR_extractor_scripts/create_unit_enum.py
#!/bin/python3
import json


with open('../Beyond-All-Reason/language/en/units.json') as f:
    data = json.load(f)

print("enum BAR_UnitType {")

for unit in data["units"]["names"]:
    print("BAR_UnitType_" + unit + ",")

print("BAR_UnitType_MAX\n};")